### PR TITLE
BUILD: Fix some cross-mingw issues

### PR DESCRIPTION
**Pull request by orgads**
* Default to `sdl2-config` instead of `sdl-config`
* Use `peldd` when cross-building
* Fix `strip`
* Do not undefine `__STRICT_ANSI__`

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -2172,7 +2173,7 @@ if test "$have_gcc" = yes ; then
 		case $_host_os in
 		# newlib-based system include files suppress non-C89 function
 		# declarations under __STRICT_ANSI__, undefine it
-		3ds | amigaos* | android | androidsdl | dreamcast | ds | gamecube | mingw* | mint* | n64 | psp | ps3 | psp2 | switch | wii )
+		3ds | amigaos* | android | androidsdl | dreamcast | ds | gamecube | mint* | n64 | psp | ps3 | psp2 | switch | wii )
```
Comment (lephilousophe) [inline configure:2176]: This change doesn't seem related to the other ones and I am not sure at all about it.

Comment (orgads) [reply configure:2176]: I had a warning on every file when `__STRICT_ANSI__` was undefined (with gcc 11 and mingw 9).

File: backends/platform/sdl/win32/win32.mk [hunk 2]
```diff
@@ -85,7 +85,11 @@ endif
 endif
 
 win32dist-mingw: win32-data
+ifeq ($(LDD),ldd)
```
Comment (lephilousophe) [inline backends/platform/sdl/win32/win32.mk:88]: I would more revert the logic here as ldd is the standard tool and peldd a specific one.
Main problem here is that you use the same variable for 2 incompatible programs

File: configure [hunk 3]
```diff
@@ -3413,10 +3414,13 @@ if test -n "$_host"; then
...
 			;;
 		*mingw32*)
-			_sdlconfig=$_host-sdl-config
+			_sdlconfig=$_host-sdl2-config
+			_libcurlconfig=$_host-curl-config
 			_windres=$_host-windres
 			_ar="$_host-ar cr"
 			_ranlib=$_host-ranlib
+			_strip=$_host-strip
+			_ldd="$_host-peldd -t --ignore-errors"
```
Comment (lephilousophe) [inline configure:3423]: This won't work if peldd is not present, and it is not on MXE.

Comment (lotharsm) [reply configure:3423]: ACK, this will break MXE.

**Comment by lotharsm** [comment]
Which issues are you trying to solve here? The MXE toolchain as well as MSYS2/Mingw-w64 representing the "other" Mingw related toolchgain work out of the box and without any additional fixes right now.

File: configure (hunk 3 above)
Comment (orgads) [reply configure:3423]: I'm using MXE, and I do have `peldd`. Do you have `pe-util` installed?

**Comment by orgads** [comment]
> Which issues are you trying to solve here? The MXE toolchain as well as MSYS2/Mingw-w64 representing the "other" Mingw related toolchgain work out of the box and without any additional fixes right now.

Does `make win32dist` and `make win32dist-mingw` work for you on MXE? For me it didn't. It tries to use `strip` instead of cross-strip, and ldd also doesn't work on PE.

File: backends/platform/sdl/win32/win32.mk (hunk 2 above)
Comment (orgads) [reply backends/platform/sdl/win32/win32.mk]: Done. Can you suggest a better alternative? Different target maybe?

File: configure (hunk 3 above)
Comment (lephilousophe) [reply configure]: I don't and it's not part of the base installation. This means you need to check if binary is present.
Something like that could be used:
`command -v "$_host-peldd" >/dev/null 2>&1 && _ldd="$_host-peldd -t --ignore-errors"` could be used.
In the same way, ldd should not be set if it isn't available.

Comment (orgads) [reply configure]: Already done :)

And it won't break MXE anyway, since ldd doesn't work either. It is only used when making the `win32dist-mingw` target, which I'm probably the only one using it :)

**Comment by lotharsm** [comment]
> Does `make win32dist` and `make win32dist-mingw` work for you on MXE? For me it didn't. It tries to use `strip` instead of cross-strip, and ldd also doesn't work on PE.

Ah, I never tried this since I use my custom build scripts. This script bascially just relies on MXE's `binutils` and `cc` targets, so I'd never had to use `peldd` or something like that.

File: configure (hunk 1 above)
Comment (lephilousophe) [reply configure:2176]: @lotharsm do you see any trouble with this?
I am not sure how it would work with older versions of mingw?
Is it newlib based after all?

Comment (lotharsm) [reply configure:2176]: @lephilousophe I don't see issues with it, and I'm curious right now why it is even in there. Looks like it _might_ only affect really ancient versions or specific builds. It might even be totally irrelevant on current versions of the C++ standard.

The only references regarding `newlib` I could find are in very old forums posts from back in 2006. MSYS2 only refers to it in conjunction with embedded ARM stuff (which is expected since newlib was created for embedded devices) and some RISCV packages [1]. I don't think newlib is used on current and "regular" MinGW instances.

I can confirm that at least on MSYS2/Mingw-w64, these changes work and don't harm anything.

To me, it looks like the `mingw*` entry here is a leftover from quite some time back ago.

[1] https://packages.msys2.org/search?q=newlib

Comment (lephilousophe) [reply configure:2176]: Hehe, as you don't know why I followed the inclusion of this in configure script.
It was there from the beginning (472ca9c68976a99802ed44dfe7399a2b11b248a5, 2004!) of mingw in configure. No mention about newlib at that time but maybe the mingw from these ancient times hide the functions if -ansi was used.
So we can assume it's not the case anymore.

**Comment by lephilousophe** [comment]
For me, it's OK.

File: configure (hunk 1 above)
Comment (ccawley2011) [reply configure:2176]: This causes a compilation failure with MinGW32 (using GCC 9.2.0 and mingwrt 5.4.2):

```
backends/fs/stdiostream.cpp: In static member function 'static StdioStream* StdioStream::makeFromPath(const Common::String&, bool)':
backends/fs/stdiostream.cpp:121:17: error: '_wfopen' was not declared in this scope; did you mean '_lopen'?
  121 |  FILE *handle = _wfopen(wPath, writeMode ? L"wb" : L"rb");
      |                 ^~~~~~~
      |                 _lopen
```

The Switch port also warns when `__STRICT_ANSI__` is undefined, but fails when removed from this case statement, so I'd suggest taking a slightly different approach.

Comment (orgads) [reply configure:2176]: The `__STRICT_ANSI__` issue is new with GCC11. See https://github.com/actions/virtual-environments/issues/3376.

I'll revert it then. Maybe later we'll revisit this and make the undefine conditional.

**Comment by orgads** [comment]
Reverted `__STRICT_ANSI__`.

**Comment by sev-** [comment]
No objections from me. @lotharsm will you merge it?

**Comment by lotharsm** [comment]
No objections from my end either. @ccawley2011, can you confirm that the issues you had with your MinGW32 toolchain is solved?

**Review by ccawley2011 — APPROVED** [review]
